pkg/controller/acmeorders/sync.go

```diff
@@ -402,6 +402,7 @@ func (c *controller) createOrder(ctx context.Context, cl acmecl.Interface, issue
 		} else {
 			cs, err = challengeSpecForAuthorization(ctx, cl, issuer, o, authz)
 			if err != nil {
+				c.recorder.Eventf(o, corev1.EventTypeWarning, "NoMatchingSolver", "Failed to create challenge for domain %q: %v", authz.Identifier.Value, err)
 				return fmt.Errorf("error constructing Challenge resource for authorization: %v", err)
```

Comment:
We should probably updated this to `return nil` in future, but thanks to #1857 we don't actually enter an endless retry loop anyway.

This can be cleaned up when #1857 is addressed too 😄